views/js/checkout-component-renderer.js

```diff
@@ -495,19 +495,19 @@ jQuery(document).ready(function() {
          * @param state
          */
         function handleOnChange(state) {
-            placeOrderAllowed = state.isValid;
+            placeOrderAllowed[state.data.paymentMethod.type] = state.isValid;
```

Comment:
Is the `state.data.paymentMethod.type` prop available  for all the payment methods? Even if the method has no fields to render like klarna (simple redirect)

Reply:
Payment methods without details won't trigger the `onChange` event anyways